### ISSUE TITLE
Request for `maxGas`

### ISSUE THREAD

**Issue by tarekskr** [trufflesuite/truffle#1910]
Truffle 5.0 brought `autoGas` and `gasMultiplier`, which are really great. It's missing however a way to set the maximum amount of gas price to pay for a transaction.

I am proposing adding a new parameter called `maxGas` that would be used as the transaction gas if the automatically computed value exceeds it.

**Comment by CruzMolina** [comment]
Great idea @tarekskr , we'll look into this!

**Comment by krsnnik** [comment]
I'd like to take a crack at this if no ones working on it!

**Comment by gnidan** [comment]
Yes please @krsnnik!

**Comment by eggplantzzz** [comment]
@krsnnik Any progress on your end on this? Let us know if you need some guidance or have any questions!

**Comment by krsnnik** [comment]
yes! @CruzMolina is helping me review in the next day or two. Should have something soon!

**Comment by gnidan** [comment]
@krsnnik taking your radio silence to mean this one's up for grabs :) Moving this to the backlog, but do let us know if you intend to work on this so someone else doesn't start working on it at the same time. Thanks!
